api/proto/teleport/accesslist/v1/accesslist.proto

```diff
@@ -285,6 +285,24 @@ message ReviewChanges {
   ReviewDayOfMonth review_day_of_month_changed = 5;
 }
 
+// MembershipOrOwnershipType describes the type of membership or ownership in an access list.
+enum MembershipOrOwnershipType {
```

Comment:
Maybe `SelfAccessType` would be a better name? In case we need to use this enum for a third field in the future that's neither about membership nor ownership.

Reply:
I named it this to keep it consistent with the existing enum from `hierarchy.go`:

https://github.com/gravitational/teleport/blob/16a4912dc654bfbf8de1b8030be64ccad885bc71/lib/accesslists/hierarchy.go#L44-L53

Do you know of a cleaner way to achieve this? It does feel a little weird having two synonymous enums

Reply:
I feel like it should be `lib/accesslists` that makes use of an enum defined in `api/proto/teleport/accesslist`, not `api` trying to mimic what's in `lib/accesslists`.

Reply:
For example, [`proto/teleport/lib/vnet/diag` defines an enum](https://github.com/gravitational/teleport/blob/9c520834d8c952c484d9e654149d89a5d4fdaa8b/proto/teleport/lib/vnet/diag/v1/diag.proto#L88-L97) that is later [used in `lib/vnet/diag`](https://github.com/gravitational/teleport/blob/9c520834d8c952c484d9e654149d89a5d4fdaa8b/lib/vnet/diag/diag.go#L99-L101) as well as [`web/packages/teleterm/src/services/vnet/diag.ts`](https://github.com/gravitational/teleport/blob/9c520834d8c952c484d9e654149d89a5d4fdaa8b/web/packages/teleterm/src/services/vnet/diag.ts#L90-L93). Protos serve as the one place that defines data structures that can be then used in different parts of the project.